app/factories/bulkrax/object_factory.rb

```diff
@@ -243,18 +243,14 @@ def update_file_set(attrs)
     def handle_remote_file(remote_file:, actor:, update: false)
       actor.file_set.label = remote_file['file_name']
       actor.file_set.import_url = remote_file['url']
+      auth_header = remote_file.fetch('auth_header', {})
 
-      url = remote_file['url']
-      tmp_file = Tempfile.new(remote_file['file_name'].split('.').first)
-      tmp_file.binmode
-
-      URI.open(url) do |url_file|
-        tmp_file.write(url_file.read)
-      end
+      ImportUrlJob.perform_now(actor.file_set, operation_for(user: @user), auth_header)
+    end
 
-      tmp_file.rewind
-      update == true ? actor.update_content(tmp_file) : actor.create_content(tmp_file, from_url: true)
-      tmp_file.close
+    def file_set_operation_for(user:)
+      Hyrax::Operation.create!(user: user,
```

Comment:
Last year, we did a lot of work to decouple Bulkrax from Hyrax. Is is fine to call `Hyrax` directly here because it's in the `ObjectFactory`?

Reply:
yes, the object factory has to be aware of Hyrax, its the one place that code is allowable.